_books/ion-1-1/src/modules/grammar.md

```diff
@@ -0,0 +1,73 @@
+# Module Grammar
+
+```bnf
+encoding-directive ::= '$ion_encoding::(' module-body ')'
+                   
+shared-module      ::= '$ion_shared_module::' ion-version-marker '::(' catalog-key module-body ')'
+
+ion-version-marker ::= '$ion_1_0' | '$ion_1_1'
+
+module-body        ::= import* inline-module* symbol-table? macro-table?
+
+import             ::= '(import ' module-name catalog-key ')'
+
+catalog-key        ::= catalog-name catalog-version?
+
+catalog-version    ::= int // positive, unannotated
+
+catalog-name       ::= string
+
+inner-module       ::= '(module' module-name import* symbol-table? macro-table? ')'
+                             
+module-name        ::= unannotated-identifier-symbol
+
+
+
+// Symbol Tables
```

Comment:
Maybe. 🤷‍♂️
The version of highlightjs we have installed doesn't seem to include `bnf`, so it doesn't really matter. IntelliJ seems to think that `//` is a comment and `;` is not, so that's what I've been doing for now. As long as people understand it, that's the main thing. I've created #355 to track cleaning up BNF snippets and adding higlighting.